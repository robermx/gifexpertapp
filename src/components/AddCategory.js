import React, { useState } from 'react';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');
  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handelSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length >= 2) {
      setCategories((cat) => [inputValue, ...cat]);
      setInputValue('');
    }
  };
  return (
    <form onSubmit={handelSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};
