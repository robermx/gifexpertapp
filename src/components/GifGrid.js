// import React, { useState, useEffect } from 'react';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="text-light animate__animated animate__fadeInDown">
        {category}
      </h3>
      {loading && (
        <p className="center animate__animated animate__pulse">Cargando</p>
      )}

      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
