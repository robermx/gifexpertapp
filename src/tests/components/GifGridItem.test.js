import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en GifGridItem', () => {
  const title = 'Un t√≠tulo';
  const url = 'https://github/hada.jpg';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test('Debe de mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de coincidir la imagen src de la url y el tutulo con alt', () => {
    const img = wrapper.find('img');
    // console.log(img.prop('src'));
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('debe de tener la clase animate__fadeInUp', () => {
    const div = wrapper.find('div');
    const animate = 'animate__fadeInUp';
    expect(div.prop('className').includes(animate)).toBe(true);
  });
});
