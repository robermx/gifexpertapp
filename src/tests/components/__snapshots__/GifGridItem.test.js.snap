// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <div
    className="card animate__animated animate__fadeInUp"
  >
    <img
      alt="Un tÃ­tulo"
      src="https://github/hada.jpg"
    />
  </div>
</Fragment>
`;
